### PR TITLE
TRUNK-4867 Check if deathdate before birthdate

### PR DESCRIPTION
**Pull request by reubenvarghese1**
## Related Issue

see https://issues.openmrs.org/browse/TRUNK-4867

## Checklist:

- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by dkayiwa** [comment]
@reubenvarghese1 can you squash these commits into one as advised at https://wiki.openmrs.org/display/docs/Pull+Request+Tips?

**Comment by reubenvarghese1** [comment]
@dkayiwa  I have done as per request.I am unable to identify the cause for the error so I do need your help with this.

**Comment by reubenvarghese1** [comment]
-

File: api/src/main/java/org/openmrs/validator/PersonValidator.java [hunk 1]
```diff
@@ -163,5 +164,19 @@ private void rejectDateIfBefore120YearsAgo(Errors errors, Date date, String date
...
+	/**
+	 * Rejects a death date if it is before birth date
+	 * 
+	 * @param errors the error object
+	 * @param deathdate the date to check
+	 * @param birthdate to check with
+	 * @param dateField the name of the field
+	 */
+	private void rejectDeathDateIfBeforeBirthDate(Errors errors, Date deathdate, Date birthdate, String dateField) {
+		if (deathdate.before(birthdate)) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/validator/PersonValidator.java]: What happens if deathdate or birthdate is null?

**Comment by dkayiwa** [comment]
You also need to some some unit tests for your changes.

File: api/src/main/java/org/openmrs/validator/PersonValidator.java (hunk 1 above)
Comment (reubenvarghese1) [inline api/src/main/java/org/openmrs/validator/PersonValidator.java]: @dkayiwa I think those conditions are already checked for in the validateBirthDate and validateDeathDate methods right?
`
private void validateBirthDate(Errors errors, Date birthDate) {
		if (birthDate == null) {
			return;
		}

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/validator/PersonValidator.java]: @reubenvarghese1 does that prevent validateDeathDate() from being called?

Comment (reubenvarghese1) [inline api/src/main/java/org/openmrs/validator/PersonValidator.java]: @dkayiwa What you say is exactly what's happening I feel.In that case I feel I'll have to make the new function totally independent of the class it's currently being called in.

Comment (reubenvarghese1) [inline api/src/main/java/org/openmrs/validator/PersonValidator.java]: What do you feel?

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/validator/PersonValidator.java]: Shouldn't a null check suffice?

Comment (reubenvarghese1) [inline api/src/main/java/org/openmrs/validator/PersonValidator.java]: I'll try what you say and get back to you 🙂. Sorry for the trouble

**Pull request closed without merging**